action.yml

```
@@ -159,8 +159,9 @@ runs:
             fi
           done | sort -u >>"$GITHUB_OUTPUT"
 
-        # Always include a top-level .stack-work
-        echo "${{ inputs.working-directory }}/.stack-work" >>"$GITHUB_OUTPUT"
+        # Always include a top-level .stack-work, with the location being driven
+        # by the stack-yaml directory.
+        echo "$(dirname ${{ format('{0}/{1}', inputs.working-directory, inputs.stack-yaml) }})/.stack-work" >>"$GITHUB_OUTPUT"
```

```suggestion
        echo "$(dirname '${{ format('{0}/{1}', inputs.working-directory, inputs.stack-yaml) }}')/.stack-work" >>"$GITHUB_OUTPUT"
```

Need a little more quoting, or `working-directory` or `stack-yaml` having spaces would've broken. I can use single-quotes here (vs double) since the `${{` interpolation becomes a literal, there's no bash variables involved.